modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/QueryMemoryTracker.java

```diff
@@ -54,47 +66,63 @@
     QueryMemoryTracker(H2MemoryTracker parent, long maxMem, long blockSize) {
         assert maxMem > 0;
 
+        // TODO GG-18629 - get from configuration.
+        failOnMemLimitExceed = !Boolean.getBoolean(IgniteSystemProperties.IGNITE_SQL_USE_DISK_OFFLOAD);
```

Comment:
This is synchronous call to HashTable.
Let's make this static.

Reply:
See explanation above.